### PR TITLE
refactor: make `Visitor` and implementations version agnostic.

### PR DESCRIPTION
**Pull request by peterhuene**
Currently, AST visitation uses `SyntaxKind` to determine which methods on a visitor to call. As `SyntaxKind` is intentionally the union of all possible WDL versions, there's no reason for the `Visitor` trait to be in a `v1` module.

This commit moves the `Visitor` trait to the root of the crate and also moves any implementations of `Visitor` out of a `v1` module as well. For validation, they now go in a `validation` module. For lint rules, they go in a `rules` module.

For the lint rules, it means that if a lint rule works only on certain nodes or tokens (e.g. whitepsace) that aren't tied to a particular version of WDL or the syntax they operate on doesn't change from version to version, the rule will continue to operate on newer versions of WDL without changes.

As new syntax is introduced over time, rules that do need to operate on the newer syntax will need to implement additional `Visitor` trait methods to visit the new syntax nodes.

If for whatever reason a rule needs to be specific to a particular version of WDL, it can implement the `document` visitor method and check the version of the document; if it's not the appropriate version, the rule can disable itself.

Before submitting this PR, please make sure:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [x] You have added tests (when appropriate).
- [x] You have updated the README or other documentation to account for these
      changes (when appropriate).
- [x] You have added an entry to the relevant `CHANGELOG.md` (see
      ["keep a changelog"] for more information).
- [x] Your commit messages follow the [conventional commit] style.

[conventional commit]: https://www.conventionalcommits.org/en/v1.0.0/#summary
["keep a changelog"]: https://keepachangelog.com/en/1.0.0/

### CODE REVIEW

**Comment by peterhuene** [comment]
I'll wait until #93 is merged and then rebase this on top of that.

**Review by a-frantz — APPROVED** [review]
Good stuff 👍

**Pull request merged**
